### PR TITLE
Push to multiple environments using single workflow

### PR DESCRIPTION
**Pull request by DrizzlyOwl**
- Fixes issue with environment not being known when triggered on `push` event
- Bumps GitHub checkout action to v3
- Adds concurrency rule to queue up deployments if multiple deploys are triggered at once
- Include `latest` tag on Docker image build to ensure it's kept up-to-date as this is the default if a tag is not specified
- Named workflow jobs to make it more human-readable
- Removed summary step as it's no longer required

### CODE REVIEW

File: .github/workflows/build-and-push-image.yml [hunk 1]
```diff
@@ -10,27 +10,42 @@ on:
...
-      - name: Set environment variables
+      - id: var
         run: |
           GIT_REF=${{ github.ref }}
           GIT_BRANCH=${GIT_REF##*/}
-          echo "BRANCH_TAG=$GIT_BRANCH" >> $GITHUB_ENV
-          echo "DOCKER_IMAGE_TAG=$GITHUB_SHA" >> $GITHUB_ENV
+          INPUT=${{ github.event.inputs.environment }}
+          ENVIRONMENT=${INPUT:-"dev"}
+          RELEASE=${ENVIRONMENT,,}-`date +%Y-%m-%d`.${{ github.run_number }}
```
Comment (jordan-beard) [inline .github/workflows/build-and-push-image.yml:35]: Out of interest what do the comma's do in this instance?

File: .github/workflows/build-and-push-image.yml [hunk 2]
```diff
@@ -45,25 +60,24 @@ jobs:
           context: .
           file: Dockerfile.gpaas-azure-migration
           tags: |
-            ${{ secrets.CIP_ACR_URL }}/${{ env.DOCKER_IMAGE }}:${{ env.DOCKER_IMAGE_TAG }}
-            ${{ secrets.CIP_ACR_URL }}/${{ env.DOCKER_IMAGE }}:${{ env.BRANCH_TAG }}
+            ${{ secrets.CIP_ACR_URL }}/${{ env.DOCKER_IMAGE }}:${{ needs.set-env.outputs.branch }}
```
Comment (jordan-beard) [inline .github/workflows/build-and-push-image.yml:63]: What is needs.set-env.outputs*?

File: .github/workflows/build-and-push-image.yml (hunk 1 above)
Comment (DrizzlyOwl) [reply .github/workflows/build-and-push-image.yml:35]: In `bash`, `$VAR^^` forces uppercase, whereas `$VAR,,` forces lowercase. This means that "Dev" becomes "dev" and "PROD" becomes "prod"

File: .github/workflows/build-and-push-image.yml (hunk 2 above)
Comment (DrizzlyOwl) [reply .github/workflows/build-and-push-image.yml:63]: The `build` job relies on a previous job called `set-env` which sets three outputs.

**outputs:**
```
environment
branch
release
```
**example values:**
```
dev
main
dev-2023-01-13.5
```

**Pull request merged**